src/helpers/augmentedReality/multipleSceneIndexGenerator.ts

```diff
@@ -80,6 +80,14 @@ export const multipleSceneIndexGenerator = ({
     }
 
     localUris = scenes?.[modelIndex]?.localUris;
+
+    if (modelIndex !== 0) {
+      const firstIndex = scenes?.[0]?.localUris;
+      const uniqueTypes = new Set(localUris.map((item) => item.type));
+
+      localUris = [...localUris, ...firstIndex.filter((item) => !uniqueTypes.has(item.type))];
```

Comment:
the commit message 

> added to localUris by finding the information in the first index for more stable operation of the AR feature

does not help me to understand this line.

can you please tell me what happens here? thanks.

Reply:
All the files we need are in the first index.

We change the `modelIndex` according to the date logic. In this case, the modelIndex we choose does not have the files we want. For example `targetImage`, `mp3` etc. We add the types in `localUris` that we use to `uniqueTypes`.

Then we filter the types that are not in `localUris[x]` and add them to `localUris[x]`. This way we have all the files we want.

Reply:
what is included in unique types and what not?

Reply:
unique types: `target`, `mp4`, `mp3`, `image`, `spotlight`, `light`, `quad`

Reply:
why aren't textures included in `const uniqueTypes = new Set(localUris.map((item) => item.type));`?

Reply:
There are `textures` and `vrx` types in `uniqueTypes`. But the types we want to add are the types I mentioned above. Therefore, we go through a filtering process.

For our date logic to work properly, we need the `textures` and `vrx` in the `x` index. We compare these types with the first index and add the types we don't have to `localUris`.

Reply:
> There are textures and vrx types in uniqueTypes. But the types we want to add are the types I mentioned above. Therefore, we go through a filtering process.

i understand that there is a filtering process and what the result should be. but i cannot understand, how this happens. can you please explain, what the set and filter does with all the localUris based on their item types!?

Reply:
this is what I did, step by step.

`scenes` is an array and we select an index from this array according to the date logic. we equate `localUris` to the `localUris` of this index. (line 82)

But there are models and textures in the index we selected. The other types we need remain in the first index.

If the `modelIndex` is greater than 0, we set the `localUris` of the first index equal to the `firstSceneLocalUris`.

In `uniqueTypes`, we match the types in the x index we select (line 86). `uniqueTypes` contains the types of models and textures. The types we need are in `firstSceneLocalUris`. But `firstSceneLocalUris` also contains models and textures. 

In line 88 we create `localUris` again. We filter and add the types that are in `uniqueTypes` on the `localUris` and not in `localUris`.

After this process, the `localUris` array consists of model, textures and other necessary file types.

Reply:
I added more descriptive text and examples to the code.

https://github.com/smart-village-solutions/smart-village-app-app/commit/90d9edfbe35db5999b228500a95de950838f9a70